internal/operators/handler/handler.go

```diff
@@ -67,6 +69,11 @@ func (h *Handler) ReportMonitoredOperatorStatus(ctx context.Context, params rest
 	if err != nil {
 		return common.GenerateErrorResponder(err)
 	}
+
+	//FIXME: should this 2 updates be in a transaction? - if yes extract the transaction of UpdateMonitoredOperatorStatus one layer above
```

Comment:
IMO it should be one transaction. Especially one case can be problematic without it:
- Update with operator status = Available;
- Monitored Operator gets updated;
- Progress update fails;

As a result the installation progress visible to the user is stuck even if the operator has completed its deployment. It will be this way until the agent retries sending the status update. At best it will be intermittent inconsistency, at worst the inconsistency will never be fixed.

Reply:
That was my thought as well, I just saw other places with some non-transcationed API calls so I was wondering - I will get it inside a transaction then.

Thanks